### PR TITLE
Add watch without selection

### LINKED ISSUE #169582: Add to watch when caret within variable name, without selection
This was originally posted here - https://github.com/microsoft/vscode-cpptools/issues/10293

When the caret is within a variable name, be able to add to watch, without having to to create a text selection (select some characters within the variable name)

1. Create new C++ project
2. Debug..
3. Place left click on variable name to place caret on / in variable.
3. Right click
4. No option to watch variable.
5. Select some characters of the variable name (even just 1)
6. Right click
7. Option to watch variable.

This is the behaviour of visual studio

1. Create new C++ project
2. Debug..
3. Place left click on variable name to place caret on / in variable.
4. Right click
5. Option to watch variable

Extension version: 1.13.8
VS Code version: Code 1.74.1 (1ad8d514439d5077d2b0b7ee64d2ce82a9308e5a, 2022-12-14T10:30:51.966Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: No

### PR DESCRIPTION
**Pull request by babakks**
Fixes #169582

I took the solution, as hinted by @roblourens, from here:
https://github.com/microsoft/vscode/blob/9fa8f02f67e2cdd82cd668c255fd5a1994a76f98/src/vs/workbench/contrib/debug/browser/debugHover.ts#L398

However, I only could test the `else` statement works in action (i.e., when `languageFeaturesService.evaluatableExpressionProvider.has(model)` returns `false`) and didn't find any way to check the `if` path's function.

### CODE REVIEW

**Review by roblourens — COMMENTED** [review]
This looks fine, but could you try to share the code with the hover rather than duplicate it? This can probably be a standalone utility helper to extract an expression from a position, it could probably be useful for some other feature someday as well.

File: src/vs/workbench/contrib/debug/browser/debugEditorActions.ts [hunk 1]
```diff
@@ -341,13 +344,35 @@ export class SelectionToWatchExpressionsAction extends EditorAction {
...
+
+		const model = editor.getModel();
+		const selection = editor.getSelection();
+
+		if (!selection.isEmpty()) {
+			expression = model.getValueInRange(selection);
+		} else {
+			const position = editor.getPosition();
+			const cancellation = new CancellationTokenSource();
+			const evaluatableExpression = await getEvaluatableExpressionAtPosition(languageFeaturesService, model, position, cancellation.token);
```
Comment (roblourens) [inline src/vs/workbench/contrib/debug/browser/debugEditorActions.ts:362]: You can actually use `CancellationToken.None` or make the cancellation token optional

Comment (babakks) [reply src/vs/workbench/contrib/debug/browser/debugEditorActions.ts]: Done. Marked the parameter as optional (with default to `CancellationToken.None`) for more convenient usages in future.

**Comment by babakks** [comment]
@roblourens We can also make the `languageFeaturesService` parameter optional (which results in the fallback one-size-fits-all path). Do you think we should do it?

**Review by roblourens — APPROVED** [review]
Thanks!

**Review by mjbvz — APPROVED** [review]

**Comment by roblourens** [comment]
> @roblourens We can also make the languageFeaturesService parameter optional (which results in the fallback one-size-fits-all path). Do you think we should do it?

I don't see why we would do that, we should use the better service whenever possible

**Pull request merged**